app/src/main/java/com/globant/weatherapp/mvp/view/WeatherView.kt

```diff
@@ -1,27 +1,34 @@
 package com.globant.weatherapp.mvp.view
 
 import android.app.Activity
+import android.view.View
 import androidx.recyclerview.widget.LinearLayoutManager
 import com.globant.weatherapp.data.entities.FiveDaysWeather
 import com.globant.weatherapp.mvp.contracts.WeatherContracts
 import com.globant.weatherapp.mvp.view.base.ActivityView
 import com.globant.weatherapp.utils.WeatherRecyclerViewAdapter
 import kotlinx.android.synthetic.main.activity_weather_info_layout.recycler_view
+import kotlinx.android.synthetic.main.activity_weather_info_layout.progress_bar
 
 class WeatherView(activity: Activity) : ActivityView(activity), WeatherContracts.View {
 
-    private lateinit var weatherAdapter: WeatherRecyclerViewAdapter
+    private var weatherAdapter: WeatherRecyclerViewAdapter = WeatherRecyclerViewAdapter()
 
-    override fun initView(weathers: FiveDaysWeather) {
-        initRecyclerView(weathers)
+    override fun initView() {
+        initRecyclerView()
+        activity?.progress_bar?.visibility = View.VISIBLE
     }
 
-    private fun initRecyclerView(weathers: FiveDaysWeather) {
+    override fun showData(weathers: FiveDaysWeather) {
+        weatherAdapter.submitList(weathers)
+        activity?.progress_bar?.visibility = View.GONE
         activity?.recycler_view?.apply {
-            layoutManager = LinearLayoutManager(this.context)
-            weatherAdapter = WeatherRecyclerViewAdapter()
             adapter = weatherAdapter
-            weatherAdapter.submitList(weathers)
+            visibility = View.VISIBLE
         }
     }
+
+    private fun initRecyclerView() {
+        activity?.recycler_view?.layoutManager = LinearLayoutManager(this.context)
```

Comment:
as this method is called only from the init view, remove this method and put this line in the initView method

Reply:
Done!